From f5dee34f17d4bb0d8752941d4bf64dfa9272b7e9 Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Thu, 16 Dec 2010 21:39:48 +0800
Subject: [PATCH 131/280] ENGR00136986 vpu: Support external RAM for searchMe
 in encoder.

searchMe will use external RAM if all IRAM for vpu is disabled.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 vpu/vpu_lib.c  | 11 +++++++++++
 vpu/vpu_lib.h  |  2 +-
 vpu/vpu_util.h |  1 +
 3 files changed, 13 insertions(+), 1 deletion(-)

diff --git a/vpu/vpu_lib.c b/vpu/vpu_lib.c
index 38d8700..ec3e77f 100644
--- a/vpu/vpu_lib.c
+++ b/vpu/vpu_lib.c
@@ -551,6 +551,11 @@ RetCode vpu_EncClose(EncHandle handle)
 	IOFreeVirtMem(&pEncInfo->picParaBaseMem);
 	IOFreePhyMem(&pEncInfo->picParaBaseMem);
 
+	/* Free searchRam if searchRam doesn't use IRAM */
+	if ((pEncInfo->secAxiUse.useHostMeEnable == 0) &&
+	    (pEncInfo->secAxiUse.searchRamAddr))
+		IOFreePhyMem(&pEncInfo->searchRamMem);
+
 	FreeCodecInstance(pCodecInst);
 	UnlockVpu(vpu_semap);
 
@@ -747,6 +752,12 @@ RetCode vpu_EncGetInitialInfo(EncHandle handle, EncInitialInfo * info)
 	/* Set secondAXI IRAM */
 	iramParam.width = pEncOP->picWidth;
 	SetEncSecondAXIIRAM(&pEncInfo->secAxiUse, &iramParam);
+	/* Use external memory if IRAM is disabled for searchMe*/
+	if (pEncInfo->secAxiUse.useHostMeEnable == 0) {
+		pEncInfo->searchRamMem.size = pEncInfo->secAxiUse.searchRamSize;
+		IOGetPhyMem(&pEncInfo->searchRamMem);
+		pEncInfo->secAxiUse.searchRamAddr = pEncInfo->searchRamMem.phy_addr;
+	}
 
 	VpuWriteReg(CMD_ENC_SEARCH_BASE, pEncInfo->secAxiUse.searchRamAddr);
 	VpuWriteReg(CMD_ENC_SEARCH_SIZE, pEncInfo->secAxiUse.searchRamSize);
diff --git a/vpu/vpu_lib.h b/vpu/vpu_lib.h
index dc583dc..8440185 100644
--- a/vpu/vpu_lib.h
+++ b/vpu/vpu_lib.h
@@ -511,7 +511,7 @@ typedef struct vpu_versioninfo {
  * v4.1.2 [2008.08.22] update MX37 VPU firmware to V1.0.5
  * v4.0.2 [2008.08.21] add the IOClkGateSet() for power saving.
  */
-#define VPU_LIB_VERSION_CODE	VPU_LIB_VERSION(5, 1, 3)
+#define VPU_LIB_VERSION_CODE	VPU_LIB_VERSION(5, 1, 4)
 
 extern unsigned int system_rev;
 
diff --git a/vpu/vpu_util.h b/vpu/vpu_util.h
index 5fba9db..87f3a7f 100644
--- a/vpu/vpu_util.h
+++ b/vpu/vpu_util.h
@@ -165,6 +165,7 @@ typedef struct {
 	EncReportInfo encReportSliceInfo;
 
 	vpu_mem_desc picParaBaseMem;
+	vpu_mem_desc searchRamMem; /* Used if IRAM is disabled */
 
 } EncInfo;
 
-- 
1.8.0

